ably/options.go

```diff
@@ -14,36 +15,72 @@ const (
 	ProtocolMsgPack = "msgpack"
 )
 
+var errInvalidKey = errors.New("invalid key format")
+
+var DefaultOptions = &ClientOptions{
+	RestEndpoint:     "https://rest.ably.io",
+	RealtimeEndpoint: "wss://realtime.ably.io:443",
+	Protocol:         ProtocolJSON, // TODO: make it ProtocolMsgPack
+}
+
 type ClientOptions struct {
-	RealtimeEndpoint string
 	RestEndpoint     string
-	Key              string
+	RealtimeEndpoint string
 	Token            string
 	Secret           string
 	ClientID         string
-	Protocol         string
+	Protocol         string // either ProtocolJSON or ProtocolMsgPack
 	UseTokenAuth     bool
-	NoTLS            bool
 
 	HTTPClient *http.Client
 }
 
-func (p *ClientOptions) Prepare() {
-	if p.Key != "" {
-		p.parseKey()
+var (
+	restTLS       = strings.NewReplacer("http://", "https://")
+	restNoTLS     = strings.NewReplacer("https://", "http://")
+	realtimeTLS   = strings.NewReplacer("ws://", "wss://", ":443", ":80")
+	realtimeNoTLS = strings.NewReplacer("wss://", "ws://", ":80", ":443")
+)
+
+func (opts *ClientOptions) SetTLS(enabled bool) {
+	if enabled {
+		opts.RestEndpoint = restTLS.Replace(opts.RestEndpoint)
+		opts.RealtimeEndpoint = realtimeTLS.Replace(opts.RealtimeEndpoint)
+	} else {
+		opts.RestEndpoint = restNoTLS.Replace(opts.RestEndpoint)
+		opts.RealtimeEndpoint = realtimeNoTLS.Replace(opts.RealtimeEndpoint)
 	}
```

Comment:
This feels like a rather cumbersome approach to what should be a simple task.  The end point should be constructed based on whether the connection is over TLS or not, along with the domain.  Why are we doing string replacing here and above when we could far more easily return an end point when requested that is constructed?